cloudevents/working-drafts/minor-versions.md

```diff
@@ -0,0 +1,337 @@
+var evt = new CloudEvent(CloudEventsSpecVersion.Latest)
+{
+  
+};
+```
+
+### Accessing attributes
+
+If an SDK exposes core attributes in a distinct way from extension attributes (as shown above,
+where `Id` is a strongly-typed property, whereas `["xyz"]` uses an indexer with a string literal)
+then any attempt to use the "special" access should fail when accessing an attribute within
+an event with a version that doesn't have that attribute as a core attribute. Example:
+
+```csharp
+// Create a v1.0 event
+var evt = new CloudEvent(CloudEventsSpecVersion.V1_0);
+
+// Try to access the new Xyz property, referring to the optional attribute introduced in
+// v1.1: this should fail.
+evt.Xyz = 1;
+
+// Access the xyz attribute by name, as an extension attribute.
+// This should work, even though the value wouldn't be valid in a v1.1 event.
+evt["xyz"] = "test";
+
+// Likewise accessing the property for an event parsed from elsewhere:
+evt = httpRequest.ParseCloudEvent();
+// If evt is v1.0, this should fail *even if it has an xyz attribute value that's valid in v1.1.
+int xyz = evt.Xyz;
+```
+
+Setting a value within an event where the version indicates that the attribute *is* known should
+validate the attribute value according to that version. Example:
+
+```csharp
+// Create a v1.1 event
+var evt = new CloudEvent(CloudEventsSpecVersion.V1_1);
+
+// This should work: 1 is a valid value, and xyz is known in v1.1
+evt.Xyz = 1;
+
+// This should fail: 123 is not a valid value for xyz in v1.1
+evt.Xyz = 123;
+
+// This should fail to compile, where that's feasible: xyz should always be an integer
+evt.Xyz = "test";
+
+// This should succeed at execution time
+evt["xyz"] = 123;
+
+// This should fail at execution time due to validation
+evt["xyz"] = "test";
+
+// This should fail at execution time due to validation
+evt["xyz"] = 123;
+```
+
+### "Changing" event versions
+
+An SDK may provide the ability to change the specversion of an event,
+or (preferred) create a new object based on the old one. When "upgrading" an event's specversion,
+the SDK should perform validation as if parsing the event from scratch. Examples:
+
+```csharp
+// Successful upgrade: no xyz value
+var evt10 = new CloudEvent(CloudEventsSpecVersion.V1_0) { Id = "..." };
+var evt11 = evt10.WithSpecVersion(CloudEventsSpecVersion.V1_1);
+
+// Successful upgrade: valid value for xyz
+var evt10 = new CloudEvent(CloudEventsSpecVersion.V1_0)
+{
+    Id = "...",
+    ["xyz"] = 10
+};
+evt11 = evt10.WithSpecVersion(CloudEventsSpecVersion.V1_1);
+
+// Failed upgrade: invalid value for xyz
+var evt10 = new CloudEvent(CloudEventsSpecVersion.V1_0)
+{
+    Id = "...",
+    ["xyz"] = "test"
+};
+evt11 = evt10.WithSpecVersion(CloudEventsSpecVersion.V1_1);
+
+// Successful downgrade
+var evt11 = new CloudEvent(CloudEventsSpecVersion.V1_1)
+{
+    Id = "...",
+    Xyz = 10
+};
+evt10 = evt11.WithSpecVersion(CloudEventsSpecVersion.V1_0);
+// Access to evt10.Xyz would fail as described above;
```

Comment:
If not an SDK defined core attribute, yes.

Reply:
The point here is the "core attributes" are not defined genetically, they are versioned in the implementations.

Reply:
I think that's an impl choice